staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -259,6 +289,43 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	})
 }
 
+// GetHistory returns the revisions associated with a StatefulSet
+func (h *StatefulSetHistoryViewer) GetHistory(namespace, name string) (map[int64]runtime.Object, error) {
+	sts, history, err := statefulSetHistory(h.c.AppsV1(), namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[int64]runtime.Object)
+	for _, h := range history {
+		applied, err := applyStatefulSetHistory(sts, h)
+		if err != nil {
+			return nil, err
+		}
+		result[h.Revision] = applied
+	}
+
+	return result, nil
+}
+
+func getDeploymentReplicaSets(apps clientappsv1.AppsV1Interface, namespace, name string) ([]*appsv1.ReplicaSet, error) {
+	deployment, err := apps.Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve deployment %s: %v", name, err)
+	}
+
+	_, oldRSs, newRS, err := deploymentutil.GetAllReplicaSets(deployment, apps)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve replica sets from deployment %s: %v", name, err)
+	}
+	allRSs := oldRSs
```

Comment:
Why we introduce `allRSs` instead using `oldRSs`?

Reply:
I just extracted the original code into a func, but didn't try to improve it.  I think you're right.  It is unnecessary to introduce this extra variable